### ISSUE TITLE
🐛 Bug Report: disable register existing component button doesn't work

### ISSUE THREAD

**Issue by GersonTrj** [backstage/backstage#26635]
### 📜 Description

I am trying to disable the register existing component button as per described in the [documentation](https://backstage.io/docs/features/software-templates/#disable-register-existing-component-button)

so I commented the part selected in the documentation:
```
const app = createApp({
  apis,
  bindRoutes({ bind }) {
    bind(catalogPlugin.externalRoutes, {
      createComponent: scaffolderPlugin.routes.root,
      viewTechDoc: techdocsPlugin.routes.docRoot,
      createFromTemplate: scaffolderPlugin.routes.selectedTemplate,
    });
    bind(apiDocsPlugin.externalRoutes, {
      registerApi: catalogImportPlugin.routes.importPage,
    });
    // bind(scaffolderPlugin.externalRoutes, {
    //     registerComponent: catalogImportPlugin.routes.importPage,
    //     viewTechDoc: techdocsPlugin.routes.docRoot,
    // });
    bind(orgPlugin.externalRoutes, {
      catalogIndex: catalogPlugin.routes.catalogIndex,
    });
  },
```

### 👍 Expected behavior

That the register component button disappears. 

### 👎 Actual Behavior with Screenshots

The register existing component button should disappear but it is still there:
<img width="508" alt="image" src="https://github.com/user-attachments/assets/e24f47c1-b093-4046-a3c9-10d73c429174">

### 👟 Reproduction steps

1. update backstage to the latest version
2. remove the commented part in app/src/App.tsx:
```
    bind(scaffolderPlugin.externalRoutes, {
        registerComponent: catalogImportPlugin.routes.importPage,
        viewTechDoc: techdocsPlugin.routes.docRoot,
    });
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by Rugvip** [comment]
Thank you for reporting! This is indeed wrong in the docs but the implementation is also somewhat incomplete, I've opened #26653 to fix both.

In the meantime, it's also possible to use the following config to explicitly disable the default route binding:

```yaml
app:
  routes:
    bindings:
      scaffolder.registerComponent: false
```

**Comment by GersonTrj** [comment]
Thanks for working on it!
I tried the config that you shared and I thought I could also do it locally so I can test the register button. In other words, I did this in my local config file:
`      scaffolder.registerComponent: true # Enables the "register existing button"
`
With the idea of having the button locally but not live. And now I get this error:
`Invalid config at app.routes.bindings['scaffolder.registerComponent'], value must be a non-empty string or false
`

I am kind of curious to understand why true is not allowed as a value =)
